lib/galaxy/webapps/galaxy/fast_app.py

```diff
@@ -109,6 +109,13 @@ async def add_x_frame_options(request: Request, call_next):
             response.headers["X-Frame-Options"] = x_frame_options
             return response
 
+        @app.middleware("http")
```

Comment:
Middlewares are evaluated on every request, can we do this in the relevant route instead ?

Reply:
Ok, I think the route now handles the cache correctly returning 304 when appropriate instead of returning the image response every time. And I increased the max-age to 30 days for now.